lib/media/media_source_engine.js

```diff
@@ -1041,8 +1041,10 @@ shaka.media.MediaSourceEngine = class {
   async endOfStream(reason) {
     await this.enqueueBlockingOperation_(() => {
       // If endOfStream() has already been called on the media source,
-      // don't call it again.
-      if (this.ended()) {
+      // don't call it again. Also do not call if readyState is
+      // 'closed' (not attached to video element) since it is not a
+      // valid operation.
+      if (this.ended() || this.mediaSource_.readyState === 'closed') {
```

Comment:
Incidentally, the `ended` method also checks the `readyState` of MS.
I considered asking you to change that instead of making the change here, but the `ended` method is used outside of this class so it's probably simpler to keep it as-is.